### PR TITLE
[#4228] improvement(client-python): Add Schema Error Handler and related exceptions, test cases in client-python

### PR DESCRIPTION
**Pull request by jingjia88**
### What changes were proposed in this pull request?

Add Schema Error Handler and related exceptions, test cases in `client-python` based on `client-java`

### Why are the changes needed?

Fix: #4228 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

UTs and ITs are added and test with `./gradlew client:client-python:test`

### CODE REVIEW

**Comment by jerryshao** [comment]
@noidname01 would you please help to review this PR?

File: clients/client-python/gravitino/exceptions/base.py [hunk 1]
```diff
@@ -93,3 +93,11 @@ class UnauthorizedException(GravitinoRuntimeException):
 
 class BadRequestException(GravitinoRuntimeException):
     """An exception thrown when the request is invalid."""
+
+
+class NoSuchCatalogException(NotFoundException):
```
Comment (noidname01) [inline clients/client-python/gravitino/exceptions/base.py:98]: I think it should be placed at same place as the `NotFoundException` and `AlreadyExistsException`, just for the readability, not a functional problem.

**Review by noidname01 — CHANGES_REQUESTED** [review]
Except for the minor issue, overall LGTM.

File: clients/client-python/gravitino/exceptions/base.py (hunk 1 above)
Comment (jingjia88) [reply clients/client-python/gravitino/exceptions/base.py]: Done!

**Review by noidname01 — APPROVED** [review]

**Pull request merged**